lib/Transforms/HandshakeHoistExtInstances.cpp

```diff
@@ -109,28 +126,49 @@ void HandshakeHoistExtInstancesPass::hoistInstances(handshake::FuncOp funcOp,
     for (auto [argNameAttr, argType] : namedArguments) {
       StringRef argName = argNameAttr.cast<StringAttr>().strref();
       resTypes.push_back(argType);
-      resNames.push_back(StringAttr::get(ctx, instName + "_" + argName));
+      resNames.push_back(StringAttr::get(ctx, instFuncName + "_" + argName));
     }
 
     // Iterate over the instance's results and add them to the function's
     // arguments
-    auto namedResults =
-        llvm::zip_equal(instFuncOp.getResNames(), instOp.getResultTypes());
-    for (auto [argNameAttr, resType] : namedResults) {
+    auto namedResults = llvm::zip_equal(
+        instFuncOp.getResNames(), instOp.getResultTypes(), instOp.getResults());
+    for (auto [argNameAttr, resType, resValue] : namedResults) {
+      // If the result goes to a sink, skip it and remove the sink op
+      auto usersRes = resValue.getUsers();
+      if (resValue.hasOneUse() && isa<handshake::SinkOp>(*usersRes.begin())) {
+        auto sinkOp = cast<handshake::SinkOp>(*usersRes.begin());
+        sinkOp.erase();
+        continue;
```

Comment:
Do you think doing this optimization will make the backend's job harder? The external function's interface is well-defined (i.e., given by the function's prototype), but we ignore a port here before going into the backend.